src/types/error.rs

```diff
@@ -82,3 +152,38 @@ pub(crate) fn convert_panics<T, F: UnwindSafe + FnOnce() -> NeonResult<T>>(f: F)
         }
     }
 }
+
+#[cfg(feature = "napi-runtime")]
+pub(crate) fn convert_panics<
+    'a,
+    T,
+    C: Context<'a> + std::panic::UnwindSafe,
+    F: UnwindSafe + FnOnce(C) -> NeonResult<T>,
+>(
+    cx: C,
+    f: F,
+) -> NeonResult<T> {
+    let env = cx.env().to_raw();
```

Comment:
This looks safe to me. The env is supposed to be valid for the duration of the function body, and the `catch_unwind` callback will only be invoked during that time.